src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -502,8 +488,14 @@ internal async Task InitializeFunctionDescriptorsAsync(IEnumerable<FunctionMetad
             Functions = functions;
         }
 
-        private void AddFunctionDescriptors()
+        private void AddFunctionDescriptors(IEnumerable<FunctionMetadata> functionMetadata)
         {
+            // Add the proxy descriptor provider if needed
+            if (functionMetadata.Any(m => m?.IsProxy ?? false))
```

Comment:
There won't (shouldn't) be any nulls in this collection. Should ensure that elsewhere. So this check should just be Any(p => p.IsProxy)